modules/nf-core/samtools/index/main.nf

```diff
@@ -11,39 +11,21 @@ process SAMTOOLS_INDEX {
     tuple val(meta), path(input)
 
     output:
-    tuple val(meta), path("*.bai") , optional:true, emit: bai
-    tuple val(meta), path("*.csi") , optional:true, emit: csi
-    tuple val(meta), path("*.crai"), optional:true, emit: crai
-    path  "versions.yml"           , emit: versions
+    tuple val(meta), path("${meta.id}_sample.bam"),     emit: bam
+    tuple val(meta), path("${meta.id}_sample.bam.bai"), emit: bai
+    path "versions.yml", emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
     """
-    samtools \\
-        index \\
-        -@ ${task.cpus} \\
-        $args \\
-        $input
+    cp "$input" "${meta.id}_sample.bam"
```

Comment:
This is an unnecessary copy, you already have the BAM or CRAM file as input so you should just merge them back together using operators